packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -182,13 +171,14 @@ export class OdspDocumentService implements IDocumentService {
                 this.epochTracker,
                 // flushCallback
                 async () => {
-                    if (this.currentConnection !== undefined && !this.currentConnection.disposed) {
-                        return this.currentConnection.flush();
+                    const currentConnection = this.odspDelayLoadedDeltaStream?.currentDeltaConnection;
```

Comment:
I think the code becomes weaker in terms of asserting claims when we use "?" all over the place.
I'd rather have a getter that asserts that this.odspDelayLoadedDeltaStream is not undefined by the time we get to needing it, and ensure that we properly load it in places where we have to.
Or alternative design (if that's the right choice) - such getter is async, but loads the chunk

Reply:
So according to the previous logic also, we were calling the flush only if current connection is defined. So, is my change changing that logic? However, I can make the change to load the delta stream, in case flush callback is invoked.

Reply:
Isn't the current logic, correct? It is not about module loading, previously we were not calling flush in case the current connection is not there which means nobody called connectTodeltaStream yet. If the current connection is there, then than means connectToDeltaStream is already called which in our case will load the module.  So, loading the module in this flush callback will be a change in behaviour.